src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/service/GovNotifyEmailService.kt

```diff
@@ -1,11 +1,59 @@
 package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service
 
+import org.slf4j.LoggerFactory
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.config.Email
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.config.EmailService
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.config.EmailTemplates
 import uk.gov.service.notify.NotificationClient
 
-internal class GovNotifyEmailService(client: NotificationClient, emailTemplates: EmailTemplates) : EmailService {
-  override fun sendEmail() {
-    TODO("Service is not yet implemented")
+class GovNotifyEmailService(
+  private val client: NotificationClient,
+  private val emailTemplates: EmailTemplates,
+) : EmailService {
+
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  override fun send(email: Email) {
+    when (email) {
+      is AppointmentCancelledCourtEmail -> send(email, emailTemplates.appointmentCancelledCourt)
+      is AppointmentCancelledPrisonEmail -> send(email, emailTemplates.appointmentCancelledPrison)
+      is AppointmentCancelledPrisonNoCourtEmail -> send(email, emailTemplates.appointmentCancelledPrisonNoCourt)
+      is OffenderReleasedCourtEmail -> send(email, emailTemplates.offenderReleasedCourt)
+      is OffenderReleasedPrisonEmail -> send(email, emailTemplates.offenderReleasedPrison)
+      is OffenderReleasedPrisonNoCourtEmail -> send(email, emailTemplates.offenderReleasedPrisonNoCourt)
+      is OffenderTransferredCourtEmail -> send(email, emailTemplates.offenderTransferredCourt)
+      is OffenderTransferredPrisonEmail -> send(email, emailTemplates.offenderTransferredPrison)
+      is OffenderTransferredPrisonNoCourtEmail -> send(email, emailTemplates.offenderTransferredPrisonNoCourt)
+      else -> throw RuntimeException("Unsupported email type ${email.javaClass.simpleName}.")
+    }
+  }
+
+  private fun send(email: Email, templateId: String) {
+    runCatching {
+      client.sendEmail(templateId, email.address, email.personalisation, null)
+    }
+      .onSuccess { log.info("EMAIL: sent ${email.javaClass.simpleName} email.") }
+      .onFailure { log.info("EMAIL: failed to send ${email.javaClass.simpleName} email.") }
+      .getOrThrow()
```

Comment:
Something to think about may be in future.

If get an exception during email sending how should the service as whole behave?  Should the transaction fail for example?

Reply:
I don't think it should fail the tx. Email notification issues should probably be alerted though. I think there might be some admin functionality we could do around resending notifications for a booking too?  But later..

Reply:
OK then this should not be propagating the exception (the `.getOrThrow()` bit)...

It should return a Result<IDENTIFIER> and the caller decides what action to take e.g. record the fact email as not sent successfully.